### PR TITLE
Replace wait-for-it image to get rid of vulnerabilities

### PR DESCRIPTION
**Pull request by marcofranssen**
This image was also outdated (no recent releases) and contains whole bunch of vulnerabilities.

### CODE REVIEW

**Comment by developer-guy** [comment]
How a cool idea this is; congrats, LGTM ✅👏

File: charts/spire/templates/agent-daemonset.yaml [hunk 1]
```diff
@@ -120,6 +125,10 @@ spec:
         - name: spire-config
           configMap:
             name: {{ include "spire.fullname" . }}-agent
+        - name: spire-wait-for-it
```
[the remark this answers is not in the document]
Comment (marcofranssen) [reply charts/spire/templates/agent-daemonset.yaml:128]: Thought this at first as well, but these are the names of the volumes within a Pod. If we change we should probably do that in the entire helm chart. On the other hand this gives more concise volume names within the pods.

**Comment by marcofranssen** [comment]
Will hold this PR while we wait for the chainguard images.

**Comment by developer-guy** [comment]
Hello, folx; I've good news: the wait-for-it image is now available on Chainguard's registry. 🥳

> cgr.dev/chainguard/wait-for-it:latest

It has been built for `amd64` architectures only, but hoping to make it more suitable with other architectures and already created a [PR](https://github.com/chainguard-images/images/pull/135) for that 🙈

/cc @dlorenc @rawlingsj

**Comment by marcofranssen** [comment]
Seems the current release doesn't have arm64 support + it has a bug.

File: charts/spire/templates/agent-daemonset.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ spec:
           # from https://github.com/lqhl/wait-for-it
```
Comment (gjkamstra) [inline charts/spire/templates/agent-daemonset.yaml:31]: Please update location: https://github.com/vishnubob/wait-for-it
```suggestion
          # from https://github.com/vishnubob/wait-for-it
```

**Review by gjkamstra — COMMENTED** [review]
wait-for-it is now coming from a different location

**Review by gjkamstra — APPROVED** [review]
LGTM

**Pull request merged**